src/components/ClaimRewards/index.tsx

```diff
@@ -0,0 +1,277 @@
+
+  const tokenType = 'UNI'
+  const amountStr = !!(tokenAmount % 1) ? tokenAmount.toFixed(2) : tokenAmount.toFixed(0)
+
+  const claim = async () => {
+    toggleShowConfetti(false)
+    setIsClaiming(true)
+
+    claimUni()
+      .then(() => {
+        setIsClaiming(false)
+        setClaimed(true)
+        toggleShowConfetti(true)
+        decrement()
+      })
+      .catch((err) => {
+        setIsClaiming(false)
+      })
+  }
+
+  if (closed) return null
+
+  return claimed || isClaiming ? (
+    <ClaimComponentWrapper>
+      <ImgWrapper>
+        <Img src={uniIcon} alt="uni-icon" />
+        {isClaiming && (
+          <CustomLightSpinner
+            style={{ position: 'absolute', top: 6, left: 155 }}
+            src={Circle}
+            alt="loader"
+            size={'90px'}
+          />
+        )}
+      </ImgWrapper>
+      <Text textAlign="center" fontWeight="500" marginTop={44}>
+        {isClaiming ? 'Claiming' : 'Claimed'}
+      </Text>
+      <Text textAlign="center" color={vars.color.genieBlue} marginTop={12} fontSize="36px" fontWeight="700">
+        {amountStr} {tokenType}
+      </Text>
+      <Text textAlign="center" marginTop={12} marginBottom={69}>
+        {isClaiming && <Text color={vars.color.grey300}> Confirm this transaction in your wallet</Text>}
+        {!isClaiming && <Text>🎉 Welcome to the Unicorn realm! 🎉</Text>}
+      </Text>
+      {!isClaiming && <CloseClaimButton onClick={() => setClosed(true)}>Close</CloseClaimButton>}
+    </ClaimComponentWrapper>
+  ) : (
+    <ClaimComponentWrapper>
+      <Body backgroundUrl={UniBackground}>
+        <Text paddingLeft={16} paddingRight={16} fontWeight={600} fontSize={16} marginTop={20}>
+          UNI Token Airdrop
+        </Text>
+        <Text paddingLeft={16} paddingRight={16} fontSize="36px" marginTop={12} fontWeight={700}>
+          {amountStr} {tokenType}
+        </Text>
+      </Body>
+      <Section>
+        <div style={{ marginBottom: 65 }}>
+          <Text lineHeight="20px" fontSize={14}>
+            As a long time supporter of Uniswap you&apos;ve been awarded {amountStr} UNI tokens used for voting and
+            governance on Uniswap. You can earn more UNI by staking your liquidity.
+          </Text>
+          <div style={{ marginTop: 12 }}>
+            <Link href="https://uniswap.org/governance">Read more about the token utility</Link>
+          </div>
+        </div>
+      </Section>
+      <ClaimButton onClick={claim}>Claim {tokenType}</ClaimButton>
+    </ClaimComponentWrapper>
+  )
+}
+
+export default function RewardsClaim() {
+  const [isOpen, setIsOpen] = useState(false)
+  const { account: address } = useWeb3React()
+  const [rewardsClaimsCount, setRewardsClaimCount] = useState(0)
+  const [showConfetti, toggleShowConfetti] = useState(false)
+  const unclaimedUniAmount: number = parseFloat(useUserUnclaimedAmount(address)?.toFixed(0) || '0')
+
+  useEffect(() => {
+    let newCount = 0
+    if (unclaimedUniAmount > 0) newCount++
```

Comment:
This component was originally written for claiming multiple rewards.  Since this was refactored to only uni, I left a lot of code that will be used for future rewards.